src/plugins/intel_gpu/tests/CMakeLists.txt

```diff
@@ -3,5 +3,6 @@
 #
 
 add_subdirectory(common)
-add_subdirectory(unit)
+# Unit tests doesn't link for some reason with icpx
```

Comment:
Please, perform the following steps to compile unit tests:
1. Merge with latest master
2. Add the following cmake lines to unit tests cmake:
```cmake
if(OV_COMPILER_IS_INTEL_LLVM)
  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=lld")
  string(REPLACE "-pie" "" CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE}")
endif()
```